backend/ibutsu_server/widgets/importance_component.py

```diff
@@ -1,64 +1,58 @@
 from collections import defaultdict
 
-from ibutsu_server.constants import BARCHART_MAX_BUILDS, JJV_RUN_LIMIT
+from sqlalchemy import desc
+
+from ibutsu_server.db.base import session
 from ibutsu_server.db.models import Result, Run
 from ibutsu_server.filters import string_to_column
-from ibutsu_server.widgets.jenkins_job_view import get_jenkins_job_view
 
 
-def get_importance_component(
-    env="prod",
-    group_field="component",
-    job_name="",
-    builds=5,
-    components="",
-    project=None,
-    count_skips=False,
-):
-    # taken from get_jenkins_line_chart in jenkins_job_analysis.py
-    run_limit = int((JJV_RUN_LIMIT / BARCHART_MAX_BUILDS) * builds)
-    jobs = get_jenkins_job_view(
-        filter_=f"job_name={job_name}",
-        page_size=builds,
-        project=project,
-        run_limit=run_limit,
-    ).get("jobs")
-
-    # A list of job build numbers to filter our runs by
-    job_ids = []
-    for job in jobs:
-        job_ids.append(job["build_number"])
-
-    # query for RUN ids
-    # metadata has to have a string to column to work
-    # because it is a sqlalchemy property otherwise (AFAIK)
+def _get_results(job_name, builds, components, project):
+    # Get the last 'builds' runs from a specific Jenkins Job as a subquery
     bnumdat = string_to_column("metadata.jenkins.build_number", Run)
-    run_data = (
-        Run.query.filter(bnumdat.in_(job_ids), Run.component.in_(components.split(",")))
-        .add_columns(Run.id, bnumdat.label("build_number"))
-        .all()
+    jnamedat = string_to_column("metadata.jenkins.job_name", Run)
+    sub_query = (
+        session.query(Run.id)
+        .filter(jnamedat == job_name)
+        .filter(Run.project_id == project)
+        .order_by(desc("start_time"))
+        .limit(builds)
+        .subquery()
     )
 
-    # get a list of the job IDs
-    run_info = {}
-    for run in run_data:
-        run_info[run.id] = run.build_number
-
+    # Filter the results based on the jenkins job, build number and component
     mdat = string_to_column("metadata.importance", Result)
+    bnumdat = string_to_column("metadata.jenkins.build_number", Result)
```

Comment:
These variable names are consistent with other modules, but are needlessly abbreviated.

No need to change them here, this is subjective. But if they bother you as much as they've bothered me, might as well make all of this read easier.

`build_num` and `job_name` would be fine variable names, maybe add a `[]_column`. All they're used for is to reference the label later on, so you could just do that here anyway and chain it.

`build_number = string_to_column("metadata.jenkins.build_number", Result).label("build_number")`

`.add_columns(<>, <>, build_number)`